It might not be related to this fix. I got another protocol error while testing. 

After 3 times move window to another display and fullscreenify, I got the following error.

![image](https://user-images.githubusercontent.com/941609/177450175-6ee562ce-5074-4422-9fb9-f99e5d50750f.png)

```
[20220706-10:52:22] [DEBUG] dynamic_monitor_data: msg_type 2 msg_length 56
[20220706-10:52:22] [DEBUG] libxrdp_process_monitor_stream: The number of monitors received is: 1
[20220706-10:52:24] [DEBUG] dynamic_monitor_data: msg_type 2 msg_length 56
[20220706-10:52:24] [DEBUG] libxrdp_process_monitor_stream: The number of monitors received is: 1
[20220706-10:52:24] [WARN ] process_dynamic_monitor_description: Not resizing. Already this size. (w: 2560 x h: 1440)
[20220706-10:52:24] [INFO ] xrdp_caps_process_pointer: client supports new(color) cursor
[20220706-10:52:25] [INFO ] xrdp_process_offscreen_bmpcache: support level 1 cache size 7864320 MB cache entries 100
[20220706-10:52:25] [INFO ] xrdp_caps_process_codecs: nscodec, codec id 1, properties len 3
[20220706-10:52:25] [WARN ] xrdp_caps_process_codecs: unknown codec id 5
[20220706-10:52:25] [INFO ] xrdp_caps_process_codecs: RemoteFX, codec id 3, properties len 49
[20220706-10:52:25] [ERROR] SSL_write: I/O error
[20220706-10:52:26] [ERROR] xrdp_sec_send_fastpath: xrdp_fastpath_send failed
[20220706-10:52:26] [ERROR] xrdp_rdp_send_fastpath: xrdp_sec_send_fastpath failed
```

--- a/xrdp/xrdp_mm.c
+++ b/xrdp/xrdp_mm.c
@@ -1009,6 +1009,29 @@ dynamic_monitor_data_first(intptr_t id, int chan_id, char *data, int bytes,
 }
 
 /******************************************************************************/
+static const int MAXIMUM_MONITOR_SIZE
+    = sizeof(struct monitor_info) * CLIENT_MONITOR_DATA_MAXIMUM_MONITORS;
+
+/******************************************************************************/
+static void
+sync_dynamic_monitor_data(struct xrdp_wm *wm,
+                          struct display_size_description *description)
+{
+    struct display_size_description *display_sizes
+        = &(wm->client_info->display_sizes);
+
+    display_sizes->monitorCount = description->monitorCount;
+    display_sizes->session_width = description->session_width;
+    display_sizes->session_height = description->session_height;
+    g_memcpy(display_sizes->minfo,
+             description->minfo,
+             MAXIMUM_MONITOR_SIZE);
+    g_memcpy(display_sizes->minfo_wm,
+             description->minfo_wm,
+             MAXIMUM_MONITOR_SIZE);
+}
+
+/******************************************************************************/
 static int
 process_dynamic_monitor_description(struct xrdp_wm *wm,
                                     struct display_size_description *description)
@@ -1039,6 +1062,18 @@ process_dynamic_monitor_description(struct xrdp_wm *wm,
             "process_dynamic_monitor_description: Not allowing resize due to"
             " invalid dimensions (w: %d x h: %d)",
             description->session_width, description->session_height);
+        return 0;
+    }
+    if (description->session_width
+            == wm->client_info->display_sizes.session_width
+            && description->session_height
+            == wm->client_info->display_sizes.session_height)
+    {
+        LOG(LOG_LEVEL_WARNING, "process_dynamic_monitor_description:"
+            " Not resizing. Already this size. (w: %d x h: %d)",
+            description->session_width,
+            description->session_height);
+        sync_dynamic_monitor_data(wm, description);
         return 0;
     }
 
@@ -1141,17 +1176,7 @@ process_dynamic_monitor_description(struct xrdp_wm *wm,
         mm->encoder = xrdp_encoder_create(mm);
     }
 
-    wm->client_info->display_sizes.monitorCount = description->monitorCount;
-    wm->client_info->display_sizes.session_width = description->session_width;
-    wm->client_info->display_sizes.session_height = description->session_height;
-    g_memcpy(wm->client_info->display_sizes.minfo,
-             description->minfo,
-             sizeof(struct monitor_info)
-             * CLIENT_MONITOR_DATA_MAXIMUM_MONITORS);
-    g_memcpy(wm->client_info->display_sizes.minfo_wm,
-             description->minfo_wm,
-             sizeof(struct monitor_info)
-             * CLIENT_MONITOR_DATA_MAXIMUM_MONITORS);
+    sync_dynamic_monitor_data(wm, description);
     return 0;
 }
 

